h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -614,6 +614,7 @@ private LazyResult queryFlat(int columnCount, ResultTarget result, long limitRow
             }
         }
         if (isForUpdateMvcc) {
+            assert forUpdateRows != null;
```

Comment:
Of course it is initialized, but IDEA struggles with that determination, and marks it as NPE-suspicious.

Reply:
There is a similar conditional code ten lines above inside a while loop. It detects it as suspicious only here?

From my point of view problems specific to Javac, JDT, code inspection in IDEA or other tools that do not lead to generation of incorrect code should not be avoided in such way. One IDE may show one incorrect warning, the next version of the same IDE may show something different, another IDE may show its own vision and so on.

The actual problem here may be in `isForUpdateMvcc` that should be copied inside a local variable to avoid potential modification during execution of this method. But not all tools are smart enough to link condition based on one variable with another variable.

Reply:
But wait, it was a leftover from initial submission with Iterator instead of Iterable and dereference forUpdateRows.iterator(), so IDEA correctly warned, because as you rightfully pointed out isForUpdateMvcc can mutate! As far as you comment about “another IDE” - realistically there is none. In 99% cases when I see it warns me - IDEA is right, even it’s not always clear to me why. 
I will remove assertion and copy flag into local variable instead.

Reply:
Oh, I forgot that actual dereference was here. There was a warning in Eclipse too, but this warning disappeared after that change. But assertions do not suppress such warnings in Eclipse.